store-api-server/src/payment/service/payment.service.ts

```diff
@@ -972,21 +975,41 @@ WHERE provider = 'simplex'
 
       events.push(event);
       for (const event1 of events) {
+        await this.createSimplexEventToDB(event1, paymentId);
         let deleteResponse = await deleteSimplexEvents(event1.event_id);
+
         if (deleteResponse?.status == 'OK') {
-          Logger.log(
-            `simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.log(`simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`);
         } else {
-          Logger.warn(
-            `simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.warn(`simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`);
         }
       }
     }
     Logger.log('getSimplexEvents FINISH successfully');
   }
 
+  private async createSimplexEventToDB(event1: any, paymentId: any) {
+    try {
+      const simplexEventCreatedResponse = await this.conn.query(
+        `
+INSERT INTO simplex_payment_event (
+  event_id, payment_id ,simplex_event_status, simplex_payment_status, 
+  partner_id, partner_end_user_id, crypto_currency, fiat_total_amount, 
+  fiat_total_amount_currency, crypto_total_amount, crypto_total_amount_currency, 
+  payment_created_at, payment_updated_at, event_timestamp
+)
+VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14 )`,
+        [event1.event_id, event1.payment.id, event1.name, event1.payment.status, event1.payment.partner_id,
```

Comment:
This approach means we will be super dependent on their event structure (what fields it has, and how they are named). As soon as they make 1 change to 1 of their fields (changing their name for example), it will cause an `undefined` insert into one of the columns of our table, and we'll silently start to fail logging this field because all columns are nullable in the `simplex_payment_event` table. This silent failure is even worse than if this code were to throw an error and fail to continue processing Simlex until fixed.

Both are undesired though. We don't want our code to start failing, neither silently nor with a thrown error, we want our code to keep logging the simplex event regardless of whether they change one of these fields or add new fields.

I think it makes sense to change the schema of `simplex_payment_event` to:
```
id SERIAL PRIMARY KEY
data JSONB NOT NULL
```

Then we can just insert the event as a json structure. And any changes from their side will still correctly be logged into our database.

Reply:
Actually, you are right. But let me think mixed solution. 
When I want to query with this table, I must think indexable fields. If all table is json the query is problem. I want to added paymentId as a index field. What do you think? @RKlompUU

Reply:
Why do you want to query on this table though? Isn't it just to store records somewhere?